pkg/app/router.go

```diff
@@ -74,25 +75,28 @@ func submitNewLinkHandler(w http.ResponseWriter, r *http.Request) {
 }
 
 func redirectHandler(w http.ResponseWriter, r *http.Request) {
+	var link util.Link
 	if r.RequestURI == "/favicon.ico" {
 		return
 	}
-	// Skip leading / in redirect link
-	id := r.RequestURI[1:]
-	link, err := fs.ReadLink(id)
+	linkID := strings.TrimLeft(r.RequestURI, "/")
+	err := fs.ReadLink(linkID, &link)
 
 	if err != nil {
```

Comment:
This condition can be inlined:

` if err := fs.ReadLink(linkID, &link); err != nil {
handleErrorWithRedirect(w, r, err)
return
}
`

Reply:
Nice, done!